### ISSUE TITLE
"unexpected token" on anonymous instance of parameterized Comparator

### ISSUE THREAD

**Issue by processing-bugs** [processing/processing#533]
_Original author: e.e.coli@gmail.com (December 24, 2010 15:22:22)_

```
import java.util.*;

final Object q = new HashMap&lt;String, Collection&lt;Integer&gt;&gt;();

final Comparator&lt;String&gt; comparator = new Comparator&lt;String&gt;()
{
  public int compare(final String value0, final String value1)
  {
    return value0.compareTo(value1);
  }
};

List&lt;String&gt; s = Arrays.asList(&quot;g&quot;, &quot;d&quot;, &quot;z&quot;); 
Collections.sort(s, comparator);
println(s);
```

_Original issue: http://code.google.com/p/processing/issues/detail?id=494_

**Comment by JakubValtar** [comment]
This works:

```processing
import java.util.*;

void setup() {
  final Object q = new HashMap<String, Collection<Integer>>();

  final Comparator<String> comparator = new Comparator<String>() {
    public int compare(final String value0, final String value1) {
      return value0.compareTo(value1);
    }
  };

  List<String> s = Arrays.asList("g", "d", "z");
  Collections.sort(s, comparator);
  println(s);
}
```

This does not:
```processing
import java.util.*;

final Object q = new HashMap<String, Collection<Integer>>();

final Comparator<String> comparator = new Comparator<String>() {
  public int compare(final String value0, final String value1) {
    return value0.compareTo(value1);
  }
};

List<String> s = Arrays.asList("g", "d", "z");
Collections.sort(s, comparator); // error on "(", error on ",",
                                 // missing a semicolon, missing right curly bracket
println(s); // error on "... VariableDeclaratorId", error on "}"
```

**Comment by benfry** [comment]
Ah, so that's a little more fixable (rather than waiting for an ANTLR4 update). Can you look into his one?

**Comment by benfry** [comment]
Let's detect this based on whether a `setup()` or `draw()` method is found.

**Comment by benfry** [comment]
Fixed in 3.0 beta 6 by @JakubValtar
